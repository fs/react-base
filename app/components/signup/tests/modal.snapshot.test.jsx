import React from 'react';
import renderer from 'react-test-renderer';
import { fakeSession } from 'mocks/fakeSession';
import SignupModal from '../modal';

jest.mock('components/modal', () => 'Modal');

describe('SignupModal', () => {
  let props;
  let component;
  const renderComponent = () => renderer.create(<SignupModal {...props} />);

  beforeEach(() => {
    props = {
      closeModal: () => { },
      isOpen: true,
      session: fakeSession,
      signupUser: () => { },
    };
  });

  it('renders correctly', () => {
    expect(renderComponent().toJSON()).toMatchSnapshot();
  });

  context('when form is invalid', () => {
    beforeEach(() => {
      component = renderComponent();

      component.getInstance().setState({
        name: 'qwe',
        email: 'qwe',
        password: 'asd',
        passwordConfirmation: 'asd',
      });
    });

    it('renders form with validation errors', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
