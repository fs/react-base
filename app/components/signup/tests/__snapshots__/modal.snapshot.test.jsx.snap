// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupModal renders correctly 1`] = `
<Modal
  closeModal={[Function]}
  isOpen={true}
  title="Sign Up"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="passwordConfirmation"
      >
        Password Confirmation
      </label>
      <input
        className="form-control"
        id="passwordConfirmation"
        name="passwordConfirmation"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</Modal>
`;

exports[`SignupModal when form is invalid renders form with validation errors 1`] = `
<Modal
  closeModal={[Function]}
  isOpen={true}
  title="Sign Up"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group has-success"
    >
      <label
        className="control-label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group has-error"
    >
      <label
        className="control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group has-error"
    >
      <label
        className="control-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-group has-error"
    >
      <label
        className="control-label"
        htmlFor="passwordConfirmation"
      >
        Password Confirmation
      </label>
      <input
        className="form-control"
        id="passwordConfirmation"
        name="passwordConfirmation"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</Modal>
`;
